### PR TITLE
[BEAM-3141][WIP] Support the coders in Python 3

### PR DESCRIPTION
**Pull request by holdenk**
This adds support for the coders in Python 3 as well as fixes some other issues that were blocking `GameStatsTest`/

### CODE REVIEW

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by asfgit** [comment]
FAILURE
 
--none--

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by jkff** [comment]
What are the next steps for this PR - is it ready for review? If so please specify reviewers.

File: sdks/python/apache_beam/coders/coder_impl.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def decode_from_stream(self, stream, nested):
       vlen = stream.read_var_int64()
       vlist = [self.decode_from_stream(stream, True) for _ in range(vlen)]
       if t == LIST_TYPE:
-        return vlist
+        return list(vlist)
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.py]: Why this change?

File: sdks/python/apache_beam/coders/coder_impl.py [hunk 2]
```diff
@@ -264,18 +264,18 @@ def get_estimated_size_and_observables(self, value, nested=False):
...
+    if value is None:
       stream.write_byte(NONE_TYPE)
     elif t is int:
       stream.write_byte(INT_TYPE)
       stream.write_var_int64(value)
     elif t is float:
       stream.write_byte(FLOAT_TYPE)
       stream.write_bigendian_double(value)
-    elif t is str:
+    elif t is bytes:
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.py]: Is bytes available in Python 2?

File: sdks/python/apache_beam/coders/coder_impl.py [hunk 3]
```diff
@@ -379,8 +381,8 @@ def _from_normal_time(self, value):
 
   def encode_to_stream(self, value, out, nested):
     span_micros = value.end.micros - value.start.micros
-    out.write_bigendian_uint64(self._from_normal_time(value.end.micros / 1000))
-    out.write_var_int64(span_micros / 1000)
+    out.write_bigendian_uint64(self._from_normal_time(old_div(value.end.micros, 1000)))
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.py]: Prefer // to a function call.

File: sdks/python/apache_beam/coders/coder_impl.py (hunk 2 above)
Comment (holdenk) [reply sdks/python/apache_beam/coders/coder_impl.py]: Yup, it's the same as string in Python 2 :)

File: sdks/python/apache_beam/coders/coder_impl.py (hunk 1 above)
Comment (holdenk) [reply sdks/python/apache_beam/coders/coder_impl.py]: So explicitly converts the generator into a list for Py3.

**Comment by holdenk** [comment]
ping @robertwb or @aaltay ? This doesn't fix everything but it gets at least a super basic pipeline working in Py3

**Comment by holdenk** [comment]
Gentle west-coast daylight hour ping :)

**Comment by aaltay** [comment]
@holdenk I was assuming that this was still a WIP PR and not paying much attention in my review list because of the PR title. Other reviewers might be doing that too. I will try to review it this or next week.

**Comment by holdenk** [comment]
Closing in favor of https://github.com/apache/beam/pull/4730

**Comment by holdenk** [comment]
Note: once the dust settles I may cherry-pick some parts of this forward in the coder area but keeping the large change up-to-date when we've switched to using six seems not overly useful.
